vendor/github.com/jcrossley3/manifestival/patch/patch.go

```diff
@@ -0,0 +1,131 @@
+package patch
+
+import (
+	"bytes"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/jsonmergepatch"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+type Patch interface {
+	IsRequired() bool
+	Merge(*unstructured.Unstructured) error
+}
+
+type jsonPatch struct {
+	patch  []byte
+	config string
+}
+
+type strategicPatch struct {
+	jsonPatch
+	schema strategicpatch.LookupPatchMeta
+}
+
+func NewPatch(src, tgt *unstructured.Unstructured) (Patch, error) {
+	var original, modified, current []byte
+	var err error
+	original = getLastAppliedConfig(tgt)
+	config := MakeLastAppliedConfig(src)
+	if modified, err = src.MarshalJSON(); err != nil {
+		return nil, err
+	}
+	if current, err = tgt.MarshalJSON(); err != nil {
+		return nil, err
+	}
+	obj, err := scheme.Scheme.New(src.GroupVersionKind())
+	switch {
+	case src.GetKind() == "ConfigMap":
```

Comment:
Yeah, I don't like special-casing ConfigMap, either. But I didn't want our lack of consensus on that matter (#163) to prevent us from fixing HEAD right now.

My goal was to make `ApplyAll` behave *exactly* like `kubectl apply` so I'm not even really keen to expose an API to override that for any type, but I concede there could be valid reasons for doing so. I just don't think this PR should be blocked by that resolution.